src/main/java/com/hazelcast/gcp/GcpClient.java

```diff
@@ -58,34 +63,59 @@
             return gcpConfig.getProjects();
         }
         LOGGER.finest("Property 'projects' not configured, fetching the current GCP project");
+
         return singletonList(RetryUtils.retry(new Callable<String>() {
             @Override
             public String call() {
-                return gcpMetadataApi.currentProject();
+                String projectFromMetadataApi = gcpMetadataApi.currentProject();
+                // Check if valid project was retrieved
+                if (projectFromMetadataApi.startsWith("<!DOCTYPE html")) {
+                    LOGGER.severe("Project name could not be retrieved. Please specify the project name in the "
+                            + "configuration if running from outside the GCP network where your hazelcast cluster is.");
+                    throw new HazelcastException("Project could not be retrieved from GCP config");
+                }
```

Comment:
1. Could you extract it to a method `validateRetrievedProject(String projectFromMetadataApi)`? And then remove the comment `// Check if valid project was retrieved`

2. Change the message to `Project name could not be retrieved. Please specify the 'projects' property if running from outside GCP network`